packages/terminal/src/node/shell-process.ts

```diff
@@ -123,3 +124,23 @@ export class ShellProcess extends TerminalProcess {
     }
     return mergedEnv;
 }
+
+/**
+ * Normalize an environment map for a given OS.
+ *
+ * On Windows it will uppercase all keys.
+ *
+ * @param env Environment variables map to normalize.
+ * @param platform Platform to normalize for.
+ * @returns New object with normalized environment variables.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function uppercaseEnv<T extends Record<string, any>>(env: T, platform: NodeJS.Platform = process.platform): T {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const normalized: any = {};
+    for (const [key, value] of Object.entries(env)) {
+        const normalizedKey = platform === 'win32' ? key.toUpperCase() : key;
```

Comment:
```suggestion

```
we can add a top if statement instead, e.g.
if ( platform !== 'win32') { 
   return env;
}

Reply:
I'll slightly change that to always return a new object.